src/components/30-organisms/carousel/index.js

```diff
@@ -0,0 +1,228 @@
+import { LitElement, html, css, unsafeCSS } from 'lit-element';
+
+/* eslint-disable import/no-extraneous-dependencies */
+import defineOnce from '../../../utils/define-once';
+import styles from './index.scss';
+import Swipe from './swipe';
+import debounce from '../../../utils/debounce';
+
+const ELEMENT_NODE = 1;
+const ANIMATION_LEFT_CLASS = 'animation-left';
+const ANIMATION_RIGHT_CLASS = 'animation-right';
+
+class AXACarousel extends LitElement {
+  static get tagName() {
+    return 'axa-carousel';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    return {
+      autorotatedisabled: { type: Boolean },
+      autorotatetime: { type: Number },
+      keysenabled: { type: Boolean },
+      _animationWrapperClass: { type: String },
+      _carouselMinHeight: { type: Number },
+    };
+  }
+
+  // public methods
+
+  handleNextButtonClick() {
+    this._nextSlide();
+    this._stopAutoRotate();
+  }
+
+  handlePreviousButtonClick() {
+    this._previousSlide();
+    this._stopAutoRotate();
+  }
+
+  // private methods
+
+  _getSlides(usedShadowRoot = this.shadowRoot) {
+    // usedShadowRoot as a property to encapsulate for unit tests
+    const slots = usedShadowRoot.querySelector('.o-carousel__slot');
+
+    return slots
+      .assignedNodes({ flatten: true })
+      .filter(node => node.nodeType === ELEMENT_NODE);
+  }
+
+  _setSlideVisibleWithAnimation(slideIndex, animationClass) {
+    this.visibleSlideIndex = slideIndex;
+    this._animationWrapperClass = '';
+    this.slides.forEach(node => {
+      node.style.display = 'none';
+    });
+
+    setTimeout(() => {
+      // Browser needs time to render css animation (>0ms). Other workarounds doesnt have effect.
+      this._animationWrapperClass = animationClass;
+      this.slides[slideIndex].style.display = 'block';
+    }, 50);
+  }
+
+  _nextSlide() {
+    let nextSlideIndex = this.visibleSlideIndex + 1;
+
+    if (nextSlideIndex > this.slides.length - 1) {
+      nextSlideIndex = 0;
+    }
+
+    this._setSlideVisibleWithAnimation(nextSlideIndex, ANIMATION_RIGHT_CLASS);
+  }
+
+  _previousSlide() {
+    let nextSlideIndex = this.visibleSlideIndex - 1;
+
+    if (nextSlideIndex < 0) {
+      nextSlideIndex = this.slides.length - 1;
+    }
+
+    this._setSlideVisibleWithAnimation(nextSlideIndex, ANIMATION_LEFT_CLASS);
+  }
+
+  _calculateContainerMinHeight() {
+    // we need to set carousel min height in case there are elements with different heights.
+    this.slides.forEach(node => {
+      node.style.display = 'block'; // not all the elements have initially a height > 0
+      if (node.clientHeight > this._carouselMinHeight) {
+        this._carouselMinHeight = node.clientHeight;
+      }
+    });
+  }
+
+  _onResize = debounce(() => {
```

Comment:
I consider this utils debounce as technical depth. I prefer do it manually with clear and set timeout as this function is an "external" dependency and goes a bit again the contained logic of components

Reply:
but up to you